### ISSUE TITLE
[FR] Show TOC of the file

### ISSUE THREAD

**Issue by dmytrodubinin** [renerocksai/telekasten.nvim#116]
**Please confirm**
- I am running the latest version of this plugin

**Description:**
It would be nice to implement the ability to see the structure (Table Of Content) of the file for quick navigation through it. For example in `wiki.vim` plugin we can generate `wikitoc` modal with file structure:
![image](https://user-images.githubusercontent.com/8066458/165579226-1934ca9c-0312-4ef9-8e5d-040ba8a83618.png)
The best option is two implementations: creating TOC directly to a file and as a pop-up window

**Comment by lambtho12** [comment]
I agree. Although there are already a few plugins for markdown editing that can do that.

That being said, this is something that could fit nicely with what @renerocksai had in mind in term of side panel. See discussion https://github.com/renerocksai/telekasten.nvim/discussions/23#discussioncomment-1754511

**Comment by lambtho12** [comment]
See #213 which is basically an extension of this idea.
